pysm/component_models/galactic/dust.py

```diff
@@ -302,3 +311,60 @@ def blackbody_ratio(freq_to, freq_from, temp) -> units.dimensionless_unscaled:
         `temp`.
     """
     return blackbody_nu(freq_to, temp) / blackbody_nu(freq_from, temp)
+
+h = const.h.value
+c = const.c.value
+k_B = const.k_B.value
+
+@njit
+def blackbody_nu(freq, temp):
+    """Calculate blackbody flux per steradian, :math:`B_{\\nu}(T)`.
+
+    .. note::
+
+        Use `numpy.errstate` to suppress Numpy warnings, if desired.
+
+    .. warning::
+
+        Output values might contain ``nan`` and ``inf``.
+
+    Parameters
+    ----------
+    in_x : number, array-like, or `~astropy.units.Quantity`
+        Frequency, wavelength, or wave number.
+        If not a Quantity, it is assumed to be in Hz.
+
+    temperature : number, array-like, or `~astropy.units.Quantity`
+        Blackbody temperature.
+        If not a Quantity, it is assumed to be in Kelvin.
+
+    Returns
+    -------
+    flux : `~astropy.units.Quantity`
+        Blackbody monochromatic flux in
+        :math:`erg \\; cm^{-2} s^{-1} Hz^{-1} sr^{-1}`.
+
+    Raises
+    ------
+    ValueError
+        Invalid temperature.
+
+    ZeroDivisionError
+        Wavelength is zero (when converting to frequency).
+
+    """
+
+    # Check if input values are physically possible
+    if np.any(temp < 0):
+        print('Temperature should be positive')
+    #if (not np.all(np.isfinite(freq))) or np.any(freq <= 0):
+    #    print('Input contains invalid wavelength/frequency value(s)')
+
+    log_boltz = h* freq * 1e9/ (k_B * temp)
+    boltzm1 = np.expm1(log_boltz)
+
+    # Calculate blackbody flux
+    bb_nu = (2.0 * h * (freq * 1e9)** 3 / (c ** 2 * boltzm1))
+    #flux = bb_nu.to(FNU, u.spectral_density(freq * 1e9))
```

Comment:
I had to import the blackbody function to optimize it with `numba` and noticed there that it is converting the output to `ergs...`. However in PySM 2 you were doing the blackbody ratio in standard units, see https://github.com/bthorne93/PySM_public/blob/master/pysm/components.py#L369. @bthorne93 can you please explain what should be the right approach?